src/pages/docs/getting-started/flutter.mdx

```diff
@@ -0,0 +1,758 @@
+</Code>
+
+The application will now display a list of clients currently present on the channel. Your current client ID should appear in the list of online users.
+
+You can have another client join the presence set using the Ably CLI:
+
+<Code>
+```shell
+ably channels presence enter my-first-channel --client-id "my-cli" --data '{"status":"From CLI"}'
+```
+</Code>
+
+## Step 4: Retrieve message history <a id="step-4"/>
+
+You can retrieve previously sent messages using the history feature. Ably stores all messages for 2 minutes by default in the event a client experiences network connectivity issues. This can be extended for longer if required.
+
+Update your `lib/widgets/messages.dart` file by finding the function `_publishMessage` and above this add the new function `_loadMessageHistory`:
+
+<Code>
+```flutter
+Future<void> _loadMessageHistory() async {
+  try {
+    // Retrieve the last 5 messages from history
+    final history = await _channel.history(
+      ably.RealtimeHistoryParams(limit: 5),
+    );
+
+    // History responses are returned in reverse chronological order (newest first)
+    // Reverse the array to show the latest messages at the bottom in the UI
+    final messagesFromHistory = history.items.reversed.toList();
+
+    setState(() {
+      _messages.clear();
+      _messages.addAll(messagesFromHistory);
+    });
+  } catch (error) {
+    debugPrint('Error loading message history: $error');
+  }
+}
+```
+</Code>
+
+Find the line `// Subscribe to messages` and above this add the following:
+
+<Code>
+```flutter
+// Load message history first
+_loadMessageHistory();
+```
+</Code>
+
+Test this feature with the following steps:
+
+1. Publish several messages using your application UI, or send messages from another client using the Ably CLI:
+
+<Code>
+```shell
+ably channels publish --count 5 my-first-channel "Message number {{.Count}}"
+```
+</Code>
+
+2. Hot reload the app or restart it. This will cause the `Messages` widget to reinitialize and call the `_loadMessageHistory()` method.
+3. You should see the last 5 messages displayed in your UI, ordered from oldest to newest at the bottom:
+
+<Code>
+```json
+Message number 1
+Message number 2
+Message number 3
+Message number 4
+Message number 5
+```
+</Code>
+
+## Step 5: Close the connection <a id="step-5"/>
+
+Connections are automatically closed approximately two minutes after the last channel is detached. However, explicitly closing connections when they're no longer needed is good practice to help save costs and clean up listeners.
+
+You can close the connection using the `AblyService`, find the line `_textController.dispose()` in the `dispose` function and below this add:
+
+<Code>
+```flutter
+// Add 10 second delay and close Ably connection
+Future.delayed(const Duration(seconds: 10), () async {
+  await AblyService().close();
+});
+```
+</Code>
+
+This ensures the connection is closed when your application terminates, freeing resources and removing listeners.
```

Comment:
This actually close connection after 10s, not when your application terminates, I would use another wording